appinventor/appengine/src/com/google/appinventor/client/ConnectProgressBar.java

```diff
@@ -0,0 +1,89 @@
+// -*- mode: java; c-basic-offset: 2; -*-
```

Comment:
I wrote the other comments in this file before this one, but as I was reviewing it I wondered why we need to have a static ConnectProgressBar INSTANCE that just wraps the functionality of the progressBar. All of its static methods just call the instance methods of the same-ish name. It seems like we could just make the ProgressBarDialogBox static, and have the static methods manipulate the static field, and do away having the separate instance of ConnectProgressBar. It seems like a level of indirection that we don't gain anything from.

Reply:
@ewpatton I want to leave the option open that other modules might want to use this feature in the future (thus the getInsance() method).